message_definitions/v1.0/common.xml

```diff
@@ -4552,6 +4552,10 @@
       <field type="uint8_t" name="increment" units="deg">Angular width in degrees of each array element.</field>
       <field type="uint16_t" name="min_distance" units="cm">Minimum distance the sensor can measure.</field>
       <field type="uint16_t" name="max_distance" units="cm">Maximum distance the sensor can measure.</field>
+      <extensions/>
+      <field type="float" name="increment_f" units="deg">Angular width in degrees of each array element as a float. If >0 then this value is used instead of uint8_t incremenet.</field>
+      <field type="float" name="angle_offset" units="deg">Relative angle offset of first distance with index 0. Value of 0 corresponding to local North/forward and matches existing map and otherwise adds an offset. Positive is to the right.</field>
+      <field type="uint8_t" name="count">Count of distances values. Valid range is 1-72. A value of 0 or >72 is treated as 72.</field>
```

Comment:
This isn't needed, it is already explicitly said in the distances array whether a distance is valid, invalid or unknown; it's already possible to send less than 72 samples.

Reply:
It solves the vague problem where you have data, then invalid, then data, then invalid, invalid... you don't know where the end is. This is to more easily allow for bad samples in variable sized packets from mixed sensors.

Reply:
How does this create variable sized packets? Extensions are always added at the end, so any of these being set means none of the array will 0 compressed, so no bandwidth is saved,

Reply:
This has nothing to do with MAVLink 2 packet length truncation. The problem is 0 is a valid distance so if you're using the packet correctly then unused values don't save bandwidth anyway. This allows the receiver to know there are less than 72 samples while allowing the UINT16_MAX sample(s) to signify that they still take up space on the arc.

Reply:
The iteration time for 72 items is really cheap on this though, so it's not saving a lot of time, and does add to the transmission overhead. (The 0 == 72 logic is also a bit painful if this is retained)

Reply:
example of a sensor that populates 5 values without the counts variable, but we don't know that it's only ever populating 5 values and the rest are always UINT16_MAX.

(treated as 5 values)
1,2,3,4,5,UINT16_MAX.....

(treated as 5 values, if we assume last valid sample is last index)
1,UINT16_MAX,3,4,5,UINT16_MAX..... (index 2 has invalid/unknown sample)

(treated as 1 sample and we lose the knowledge that there are 4 more zones of invalid data)
1,UINT16_MAX,UINT16_MAX,UINT16_MAX,UINT16_MAX,UINT16_MAX..... (index 2,3,4,5 have invalid/unknown sample).

Reply:
I talked it over with @jmachuca77 and, even though this help would create less work on the autopilot (receiving) side athe cost of just one transport byte, the end result distance mapping would likely always be the same. So, I'll remove COUNT entry. All Hail STM32F7!